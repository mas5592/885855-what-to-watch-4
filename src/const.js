export const ENTRY_POINT = `https://4.react.pages.academy/wtw`;

export const AVATAR_URL = `https://4.react.pages.academy/wtw/asda.jpg`;

export const TIMEOUT_DURATION = 5000;

export const MAX_SIMILAR_CARDS = 4;

export const LIMIT_FILMS_COUNT = 8;

export const RATINGS_NUMBER = 5;

export const FILTER_ALL_GENRES = `All genres`;

export const PageNames = {
  MAIN: `main`,
  FILM_DETAILS: `Details`,
  SIGN_IN: `sign in`,
  ADD_REVIEW: `add review`,
  MY_LIST: `my list`,
};

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  PAGE: `/films`,
  FILM_REVIEW: `/review`,
  MY_LIST: `/mylist`,
  VIDEO_PLAYER: `/player`,
};

export const RatingLevel = {
  AWESOME: `Awesome`,
  VERY_GOOD: `Very good`,
  GOOD: `Good`,
  NORMAL: `Normal`,
  BAD: `Bad`
};

export const ListFilters = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};

export const Time = {
  MINUTES_IN_HOUR: 60,
  SECONDS_IN_MINUTE: 60,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const ReviewLength = {
  MIN: 50,
  MAX: 400,
};

export const reviewSubmitBtn = {
  post: `Post`,
  sending: `Sending...`,
};
