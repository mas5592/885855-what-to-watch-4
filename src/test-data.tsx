import {FilmType, ReviewType} from './types';

export const card: FilmType = {
  title: `Name`,
  genre: `film Genre`,
  date: 1993,
  coverBackground: `https://placeimg.com/1300/512/nature`,
  poster: `img/snatch.jpg`,
  picture: `img/snatch.jpg`,
  id: 123456,
  description: `Place to describe the movie`,
  rating: 8.9,
  scores: 240,
  director: `Director Name`,
  starring: [`Author`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  filmDurationTime: 107,
  videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  isFavorite: false,
  backgroundColor: `#FFFFFF`,
};

export const films: Array<FilmType> = [
  {
    title: `Name`,
    genre: `Genre`,
    date: 2014,
    coverBackground: `https://placeimg.com/1300/512/nature`,
    poster: `img/snatch.jpg`,
    picture: `img/snatch.jpg`,
    id: 123465,
    description: `Place to describe the movie`,
    rating: 8.3,
    scores: 1500,
    director: `Director Name`,
    starring: [`Author`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    filmDurationTime: 107,
    videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    isFavorite: false,
    backgroundColor: `#FFFFFF`,
  },
  {
    title: `Name`,
    genre: `Genre`,
    date: 2014,
    coverBackground: `https://placeimg.com/1300/512/nature`,
    poster: `img/snatch.jpg`,
    picture: `img/snatch.jpg`,
    id: 123478,
    description: `film description.`,
    rating: 8.3,
    scores: 1500,
    director: `Director Name`,
    starring: [`Author`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    filmDurationTime: 107,
    isFavorite: false,
    backgroundColor: `#FFFFFF`,
  }
];

export const activeFilm: FilmType = {
  title: `Snatch`,
  genre: `Genre`,
  date: 2014,
  coverBackground: `https://placeimg.com/1300/512/nature`,
  poster: `img/snatch.jpg`,
  picture: `img/snatch.jpg`,
  id: 123891,
  description: `film description.`,
  rating: 8.3,
  scores: 1500,
  director: `Director Name`,
  starring: [`Author`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  filmDurationTime: 107,
  videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  isFavorite: false,
  backgroundColor: `#FFFFFF`,
};

export const reviews: Array<ReviewType> = [{
  id: 1234567891,
  user: {
    id: 1223,
    name: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment.`,
  date: `2020-07-09T10:12:45.467Z`,
},
{
  id: 1234567891,
  user: {
    id: 1223,
    name: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment`,
  date: `2020-07-09T10:12:45.467Z`,
},
{
  id: 1234567891,
  user: {
    id: 1223,
    name: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment`,
  date: `2020-07-09T10:12:45.467Z`,
}];

export const genres: Array<string> = [
  `All genres`,
  `Drama`,
  `Thriller`,
  `Comedy`
];
