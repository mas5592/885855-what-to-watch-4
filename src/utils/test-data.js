export const card = {
  title: `Name`,
  genre: `film Genre`,
  date: 1993,
  coverBackground: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  picture: `img/snatch.jpg`,
  id: 123456,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege`,
  rating: 8.9,
  ratingDescription: `Very good`,
  scores: 240,
  director: `Director Name`,
  starring: [`Author`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/7/72/Landwasserviadukt%2C_aerial_video.webm`,
  filmDurationTime: 107,
  videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  isFavorite: false,
  backgroundColor: `#FFFFFF`,
};

export const films = [
  {
    title: `Snatch`,
    genre: `Genre`,
    date: 2014,
    coverBackground: `https://placeimg.com/1300/512/nature`,
    poster: `img/snatch.jpg`,
    picture: `img/snatch.jpg`,
    id: 123465,
    description: `film description.`,
    rating: 8.3,
    scores: 1500,
    director: `Director Name`,
    starring: [`Author`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    filmDurationTime: 107,
    videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    isFavorite: false,
    backgroundColor: `#FFFFFF`,
  }, {
    title: `Snatch`,
    genre: `Genre`,
    date: 2014,
    coverBackground: `https://placeimg.com/1300/512/nature`,
    poster: `img/snatch.jpg`,
    picture: `img/snatch.jpg`,
    id: 123478,
    description: `film description.`,
    rating: 8.3,
    scores: 1500,
    director: `Director Name`,
    starring: [`Author`],
    preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    filmDurationTime: 107,
    videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
    isFavorite: false,
    backgroundColor: `#FFFFFF`,
  }
];

export const activeFilm = {
  title: `Snatch`,
  genre: `Genre`,
  date: 2014,
  coverBackground: `https://placeimg.com/1300/512/nature`,
  poster: `img/snatch.jpg`,
  picture: `img/snatch.jpg`,
  id: 123891,
  description: `film description.`,
  rating: 8.3,
  scores: 1500,
  director: `Director Name`,
  starring: [`Author`],
  preview: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  filmDurationTime: 107,
  videoUrl: `https://upload.wikimedia.org/wikipedia/commons/1/19/Ziteil%2C_aerial_video.webm`,
  isFavorite: false,
  backgroundColor: `#FFFFFF`,
};

export const filmReviews = [{
  id: 1234567891,
  user: {
    id: 1223,
    title: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment`,
  date: `2020-07-09T10:12:45.467Z`,
},
{
  id: 1234567891,
  user: {
    id: 1223,
    title: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment`,
  date: `2020-07-09T10:12:45.467Z`,
},
{
  id: 1234567891,
  user: {
    id: 1223,
    title: `Ivan Ivanov`,
  },
  rating: 8.0,
  comment: `My comment`,
  date: `2020-07-09T10:12:45.467Z`,
}];

export const genres = [`All genres`, `Drama`, `Thriller`, `Comedy`];
