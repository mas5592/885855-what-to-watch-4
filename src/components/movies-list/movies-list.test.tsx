import * as React from 'react';
import * as renderer from 'react-test-renderer';
import MoviesList from './movies-list';
import history from '../../history';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
import {FILTER_ALL_GENRES} from '../../const.js';
import {films} from '../../test-data';
import {Router} from 'react-router-dom';


const mockStore = configureStore([]);

describe(`MoviesList`, () => {
  it(`Should render MoviesList component`, () => {
    const store = mockStore({
      films,
      activeGenre: FILTER_ALL_GENRES,
    });

    const tree = renderer
      .create(
          <Router history={history}>
            <Provider store={store}>
              <MoviesList
                films={films}
                render={() => null}
              /></Provider>
          </Router>, {
            createNodeMock: () => {
              return {};
            }
          }
      )
  .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
