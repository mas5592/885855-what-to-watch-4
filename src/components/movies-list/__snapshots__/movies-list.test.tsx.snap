// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesList Should render MoviesList component 1`] = `
<div
  className="catalog__movies-list"
>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/films/123465"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          className="player__video"
          poster="img/snatch.jpg"
        />
        <img
          alt="Name"
          height={175}
          src="img/snatch.jpg"
          width={280}
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        Name
      </h3>
    </a>
  </article>
  <article
    className="small-movie-card catalog__movies-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="/films/123478"
      onClick={[Function]}
    >
      <div
        className="small-movie-card__image"
      >
        <video
          className="player__video"
          poster="img/snatch.jpg"
        />
        <img
          alt="Name"
          height={175}
          src="img/snatch.jpg"
          width={280}
        />
      </div>
      <h3
        className="small-movie-card__title"
      >
        Name
      </h3>
    </a>
  </article>
</div>
`;
